src/synapse_api.py

```diff
@@ -40,139 +38,109 @@ def __init__(self, msg: str):
-
-
-class SynapseAPI:
-    """The Synapse API handler."""
-
-    def __init__(self, charm: CharmBase):
-        """Initialize the handler.
 
-        Args:
-            charm: The parent charm to attach the observer to.
-        """
-        self._charm = charm
+class NetworkError(SynapseAPIError):
+    """Exception raised when requesting API fails due network issues."""
 
-    def _get_nonce(self) -> str:
-        """Get nonce.
 
-        Returns:
-            The nonce returned by Synapse API.
+def _get_nonce() -> str:
+    """Get nonce.
 
-        Raises:
-            NetworkError: if there was an error fetching the nonce.
-        """
-        try:
-            res = requests.get(URL_REGISTER, timeout=5)
-            res.raise_for_status()
-            return res.json()["nonce"]
-        except (
-            requests.exceptions.ConnectionError,
-            requests.exceptions.Timeout,
-            requests.exceptions.HTTPError,
-        ) as exc:
-            logger.error("Failed to request %s : %s", URL_REGISTER, exc)
-            raise NetworkError(f"Failed to request {URL_REGISTER}.") from exc
-
-    def _generate_mac(
-        self,
-        shared_secret: str,
-        nonce: str,
-        user: str,
-        password: str,
-        admin: bool = False,
-        user_type: typing.Optional[str] = None,
-    ) -> str:
-        """Generate mac to register user.
+    Returns:
+        The nonce returned by Synapse API.
 
-        Args:
-            shared_secret: registration_shared_secret from configuration file.
-            nonce: nonce generated.
-            user: username.
-            password: password.
-            admin: if is admin. Default False.
-            user_type: user type. Defaults to None.
-
-        Returns:
-            _type_: _description_
-        """
-        mac = hmac.new(key=shared_secret.encode("utf8"), digestmod=hashlib.sha1)
-        mac.update(nonce.encode("utf8"))
-        mac.update(b"\x00")
-        mac.update(user.encode("utf8"))
-        mac.update(b"\x00")
-        mac.update(password.encode("utf8"))
+    Raises:
+        NetworkError: if there was an error fetching the nonce.
+    """
+    try:
+        res = requests.get(REGISTER_URL, timeout=5)
+        res.raise_for_status()
+        return res.json()["nonce"]
+    except (
+        requests.exceptions.ConnectionError,
+        requests.exceptions.Timeout,
+        requests.exceptions.HTTPError,
+    ) as exc:
+        logger.error("Failed to request %s : %s", REGISTER_URL, exc)
+        raise NetworkError(f"Failed to request {REGISTER_URL}.") from exc
+
+
+def _generate_mac(
+    shared_secret: str,
+    nonce: str,
+    user: str,
+    password: str,
+    admin: bool = False,
+    user_type: typing.Optional[str] = None,
+) -> str:
+    """Generate mac to register user.
+
+    Args:
+        shared_secret: registration_shared_secret from configuration file.
+        nonce: nonce generated.
+        user: username.
+        password: password.
```

Comment:
Perhaps expand on this a bit? Describing the argument using the name of the argument doesn't add much value